src/web/pages/usersettings/usersettingspage.jsx

```diff
@@ -237,23 +237,45 @@ class UserSettings extends React.Component {
     }
   }
 
-  handleSaveSettings(data) {
-    const {gmp} = this.props;
-    const {userInterfaceLanguage = BROWSER_LANGUAGE, timezone} = data;
+  async handleSaveSettings(data) {
+    try {
+      const {gmp} = this.props;
 
-    this.handleInteraction();
+      const {userInterfaceLanguage = BROWSER_LANGUAGE, timezone} = data;
 
-    return gmp.user.saveSettings(data).then(() => {
-      this.closeDialog();
-      this.props.setLocale(
-        userInterfaceLanguage === BROWSER_LANGUAGE
-          ? undefined
-          : userInterfaceLanguage,
+      this.handleInteraction();
+      await gmp.user.saveSetting(
+        'd9857b7c-1159-4193-9bc0-18fae5473a69',
+        data.userInterfaceDateFormat,
+      );
+      await gmp.user.saveSetting(
+        '11deb7ff-550b-4950-aacf-06faeb7c61b9',
+        data.userInterfaceTimeFormat,
       );
```

Comment:
You may want to remove the explicit `saveSetting` here and instead let `saveSettings` handle the saving. Since this will require the `gsad` updates, the changes in this PR would need to be ported to gsa main as well.
```suggestion
```